### PR TITLE
feat: Add usage with Tailwind documentation

### LINKED ISSUE #64: Add documentation on usage with Tailwind CSS
I've been a user of this library for a long time and took me some time to figure out how to enable Tailwind IntelliSense inside `clsx` functions. I think could be helpful to add documentation on how to enable it to the README.

I'll open a PR

### PR DESCRIPTION
**Pull request by edgarlr**
Add documentation on how to enable Tailwind IntelliSense inside `clsx` functions

Fixes #64

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -83,6 +83,28 @@ All browsers that support [`Array.isArray`](https://developer.mozilla.org/en-US/
 
 - [obj-str](https://github.com/lukeed/obj-str) - A smaller (96B) and similiar utility that only works with Objects.
 
+## Tailwind CSS
```
Comment (lukeed) [inline readme.md:86]:
```suggestion
## Tailwind Support

```

**Review by lukeed — COMMENTED** [review]
Thanks! Can you move this section between `Support` and `Related`?

**Comment by edgarlr** [comment]
@lukeed Done!

**Comment by lukeed** [comment]
Thanks :)

File: readme.md [hunk 2]
```diff
@@ -78,6 +78,27 @@ All browsers that support [`Array.isArray`](https://developer.mozilla.org/en-US/
...
+</summary>
+
+1. [Install the "Tailwind CSS IntelliSense" Visual Studio Code extension](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss)
+
+2. Add the following to your [`settings.json`](https://code.visualstudio.com/docs/getstarted/settings):
+
+  ```json
+   {
+    "tailwindCSS.experimental.classRegex": [
+      ["clsx\\(([^)]*)\\)", "(?:'|\"|`)([^']*)(?:'|\"|`)"]
```
Comment (kachkaev) [inline readme.md:97]: 👋 @lukeed! Thanks for adding this regex! Can you please elaborate on your approach to building it? I was using https://github.com/shadcn/ui/blob/5e172fc34fdf015aa0d141f52cc8c082b8ae6613/.vscode/settings.json#L7-L8 before seeing your solution. There is a slight difference, so I'm curious if there are any edge cases not covered by any of them:

```diff
shadcn/ui
- ["cn\\(([^)]*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]"]
+ ["cn\\(([^)]*)\\)", "(?:'|\"|`)([^']*)(?:'|\"|`)"]
your regex
```

Comment (lukeed) [reply readme.md:97]: @edgarlr contributed this section so perhaps they can offer an explanation.

By eyeballing it, the difference is that `([^']*)` will match anything up to `'`, which means that if `"` or or backticks appear in the class value, then the regexp in _our_ docs will include it as part of the value, whereas the other regexp stops there.

clsx doc's regexp
Screen Shot 2023-07-18 at 10 43 53 AM

shadcn/ui regexp
Screen Shot 2023-07-18 at 10 44 16 AM

Looks like they're both not ideal, tbh
I don't use this extension, but perhaps something like this is better? I don't know if the extn is looking for the first matching group only, though

Screen Shot 2023-07-18 at 10 53 57 AM

**Pull request merged**